### PR TITLE
chore: Simpler Sharing Page [Concept] 

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
Simplifies the sharing page by using common component of analysis pages.

Fixes
https://linear.app/formbricks/issue/FOR-1977/[concept]-simpler-sharing-page

## How should this be tested?

- Create a survey and make the results public, then just view the results page in incognito, it should work without any error.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Review by pandeymangg — COMMENTED** [review]
Hey @gupta-piyush19 Thanks for the PR, looks good and works well 😊 
I've just added a small check to the `useMembershipRole` hook to not call the api on the sharing page, other than that, LGTM!

File: packages/lib/membership/hooks/useMembershipRole.tsx [hunk 1]
```diff
@@ -10,12 +10,14 @@ enum MembershipRole {
   Viewer = "viewer",
 }
 
-export const useMembershipRole = (environmentId: string) => {
+export const useMembershipRole = (environmentId: string, isSharingPage: boolean = false) => {
```
[the remark this answers is not in the document]
Comment (mattinannt) [reply packages/lib/membership/hooks/useMembershipRole.tsx:13]: Maybe we shouldn't use the generic hook in the responses/summary page but call the `getMembershipByUserIdTeamIdAction` action manually & conditionally

**Review by mattinannt — CHANGES_REQUESTED** [review]
@gupta-piyush19 Looks good :-) I only have one concern regarding the modification of the generic `useMembershipRole` hook:

**Review by mattinannt — APPROVED** [review]
@gupta-piyush19 thanks for the changes; looks good :-) 
Please go ahead with the merge :-)

**Pull request merged**